lib/src/arithmetic/arithmetic_utils.dart

```diff
@@ -0,0 +1,112 @@
+// Copyright (C) 2024 Intel Corporation
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// floating_point_test.dart
+// Tests of Floating Point stuff
+//
+// 2024 August 30
+// Author: Desmond A Kirkpatrick <desmond.a.kirkpatrick@intel.com
+
+// ignore_for_file: avoid_print
+
+import 'package:rohd/rohd.dart';
+import 'package:rohd_hcl/rohd_hcl.dart';
+
+/// Helper evaluation methods for printing aligned arithmetic bitvectors.
+extension NumericVector on LogicValue {
+  /// Print aligned bitvector with an optional header.
+  /// [name] is printed at the LHS of the line, trimmed by [prefix].
+  /// [prefix] is the distance from the margin bebore the vector is printed.
+  /// You can align with longer bitvectors by stating the length [align].
+  /// [lowLimit] will trim the vector below this bit position.
+  /// You can insert a separator [sepChar] at position [sep].
+  /// A header can be printed by setting [header] to true.
+  /// Markdown format can be produced by setting [markDown] to true.
+  String vecString(String name,
+      {int prefix = 10,
+      int? align,
+      int? sep,
+      bool header = false,
+      String sepChar = '*',
+      int lowLimit = 0,
+      bool markDown = false}) {
+    final str = StringBuffer();
+    // ignore: cascade_invocations
+    if (header) {
+      str.write(markDown ? '|Name' : ' ' * prefix);
+
+      for (var col = ((align ?? width) - width) + width - 1;
+          col >= lowLimit;
+          col--) {
+        final bits = col > 9 ? 2 : 1;
```

Comment:
log(col)?  what about >=100 bits?

Reply:
The format is currently limited because of the way I print headers.
Should I fix, or throw?  Fixing will slow the PR and it is a debug routine.

I also want to merge the production of ASCII and markdown so that the logic is not replicated.  I did this in my most
recent vecString routine.

Reply:
since it's for debug it's probably fine to leave as-is.  throwing an exception might be nice, or even just adding a comment that it "only works for <100 bits" or something would work too

Reply:
I decided to fix.  I want to use this routine in ALL my evaluate routines because it is something I do all the time.
That will give me Markdown and potentially HTML forms of these printouts through reuse.  I have two other routines that print:  the partial product generator  and the compression tree:  we will be looking at both of these to debug VMM
core.